packages/network-controller/src/NetworkController.ts

```diff
@@ -756,18 +762,97 @@ export class NetworkController extends BaseControllerV2<
     return convertNetworkId(possibleNetworkId);
   }
 
+  /**
+   * Refreshes the network meta with EIP-1559 support and the network status
+   * based on the given network client ID.
+   *
+   * @param networkClientId - The ID of the network client to update.
+   */
+  async lookupNetworkByClientId(networkClientId: NetworkClientId) {
+    const isInfura = networkClientId in InfuraNetworkType;
```

Comment:
`isInfuraProviderType(networkClientId)`? not sure if these are equivalent or not